brushfire-scalding/src/main/scala/com/stripe/brushfire/scalding/Trainer.scala

```diff
@@ -89,7 +90,7 @@ case class Trainer[K: Ordering, V, T: Monoid](
             case (treeIndex, map) => {
               val newTree =
                 treeMap(treeIndex)
-                  .updateByLeafIndex { index => map.get(index).map { t => LeafNode(index, t) } }
+                  .updateByLeafIndex { index => map.get(index).map { t => LeafNode(index, t, Monoid.zero[A]) } }
```

Comment:
Do we need the annotations during training? May be easier to fix `A` to `Unit` or something and just avoid the whole thing. If this doesn't work with what you are doing, then no worries... we'll have to figure something out.